### PR TITLE
Assorted fixture and API improvements

### PR DESCRIPTION
**Pull request by jerith**

### CODE REVIEW

File: seaworthy/tests-core/containers/test_base.py [hunk 1]
```diff
@@ -87,3 +87,22 @@ def test_default_clean(self):
...
+            '8080/tcp': ('127.0.0.1',),
+            '9090/tcp': ('127.0.0.1', '10701'),
+        }}
+        dh = self.make_helper()
+        self.base.create_and_start(dh, pull=False)
+
+        # We get a random high port number here.
+        random_host_port = self.base.get_host_port('8080/tcp')
+        self.assertGreater(int(random_host_port), 1024)
+        self.assertLess(int(random_host_port), 102400)
```
Comment (JayH5) [inline seaworthy/tests-core/containers/test_base.py]: I thought the highest port number was 65535?

Comment (jerith) [reply seaworthy/tests-core/containers/test_base.py]: Probably. I was lazy about choosing a number.

File: seaworthy/tests-core/containers/test_base.py [hunk 2]
```diff
@@ -87,3 +87,22 @@ def test_default_clean(self):
...
+        dh = self.make_helper()
+        self.base.create_and_start(dh, pull=False)
+
+        # We get a random high port number here.
+        random_host_port = self.base.get_host_port('8080/tcp')
+        self.assertGreater(int(random_host_port), 1024)
+        self.assertLess(int(random_host_port), 65536)
+
+        # We get the specific port we defined here.
+        self.assertEqual(self.base.get_host_port('9090/tcp', 0), '10701')
```
Comment (JayH5) [inline seaworthy/tests-core/containers/test_base.py]: Should maybe add a test for index > 0?

Comment (jerith) [reply seaworthy/tests-core/containers/test_base.py]: I had one, but the ordering of the mapped ports doesn't match the input order I used for a couple of examples and I don't know what (if any) rules that might follow.

I'm kind of tempted to just not support multiple mappings for a port. This is a testing tool, so I don't think multiple port mappings are necessarily useful.

File: seaworthy/containers/provided.py [hunk 3]
```diff
@@ -116,6 +118,7 @@ def __init__(self,
                  name=DEFAULT_NAME,
                  image=DEFAULT_IMAGE,
                  wait_patterns=DEFAULT_WAIT_PATTERNS,
+                 create_kwargs=None,
```
Comment (JayH5) [inline seaworthy/containers/provided.py]: This seems unnecessary?

Comment (jerith) [reply seaworthy/containers/provided.py]: We might want to pass a command to run or some default envvars or whatever in at construction time but then pass different things in when the container is created.

Comment (JayH5) [reply seaworthy/containers/provided.py]: Ah, think I misread the diff.

File: seaworthy/tests-core/test_logs.py [hunk 4]
```diff
@@ -179,8 +179,12 @@ def attach_socket(self, params):
         server, client = socket.socketpair()
         self._client_sockets.add(client)
         self._feeder = LogFeeder(self, server)
+        fileobj = socket.SocketIO(client, 'rb')
+        # Add a "response" object for the streaming client close. We do this
```
Comment (JayH5) [inline seaworthy/tests-core/test_logs.py]: "streaming client _to_ close"?

File: seaworthy/tests-core/test_logs.py [hunk 5]
```diff
@@ -179,8 +179,12 @@ def attach_socket(self, params):
...
+        # before starting the feeder to avoid races with really fast logs.
+        fileobj._response = self._feeder
```
Comment (JayH5) [inline seaworthy/tests-core/test_logs.py]: Setting a private attribute on an undocumented class 👏 

...it does clean things up a fair bit, though.

**Pull request merged**